### PR TITLE
Get the backstack working.

### PR DESCRIPTION
**Pull request by jimmyfnugent**
Filters are now updated when you hit forward/back.

### CODE REVIEW

File: client/src/Home.tsx [hunk 1]
```diff
@@ -19,6 +20,10 @@ const mapStateToProps = (state: AppState, ownProps: InputProps) => ({
 class Home extends AppRouteComponent<typeof mapStateToProps> {
   componentDidMount() {
     this.props.thunkSetStateFromUrlBar();
+    history.listen((location, action) => {
```
Comment (hopskipnfall) [inline client/src/Home.tsx:23]: We probably need to unsubscribe from this somehow when the page is destructed.. This callback will forever be called even when we're not on Home anymore.

File: client/src/fake_api_client.ts [hunk 2]
```diff
@@ -107,34 +107,45 @@ export const fakeApiClient = {
     const filtered = bits.filter((bit) => {
       if (filters.mainCharacters.size > 0) {
         if (!bit.mainChars) return false;
-        for (const char of Array.from(filters.mainCharacters)) {
-          if (!bit.mainChars || bit.mainChars.indexOf(char.id) === -1)
-            return false;
+        if (
+          Array.from(filters.mainCharacters).filter(
```
Comment (hopskipnfall) [inline client/src/fake_api_client.ts:111]: use `.find()` instead of `.filter().length` (throughout)

**Review by hopskipnfall — CHANGES_REQUESTED** [review]
what happened to our presubmits..

File: client/src/Home.tsx (hunk 1 above)
Comment (jimmyfnugent) [reply client/src/Home.tsx:23]: Done.

File: client/src/fake_api_client.ts (hunk 2 above)
Comment (jimmyfnugent) [reply client/src/fake_api_client.ts:111]: Done.

**Pull request merged**